<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.37">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="AJ Smit">

<title>Lab 1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>


<script src="01-lab_1_files/libs/clipboard/clipboard.min.js"></script>
<script src="01-lab_1_files/libs/quarto-html/quarto.js"></script>
<script src="01-lab_1_files/libs/quarto-html/popper.min.js"></script>
<script src="01-lab_1_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="01-lab_1_files/libs/quarto-html/anchor.min.js"></script>
<link href="01-lab_1_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="01-lab_1_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="01-lab_1_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="01-lab_1_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="01-lab_1_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Lab 1</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>AJ Smit </p>
          </div>
  </div>
    
    
  </div>
  

</header>

<blockquote class="blockquote">
<p><em>“A scientific man ought to have no wishes, no affections, – a mere heart of stone.”</em></p>
<p>— Charles Darwin</p>
</blockquote>
<section id="macroecology" class="level2">
<h2 class="anchored" data-anchor-id="macroecology">Macroecology</h2>
<p><strong>This is a course about community ecology across different spatial [and temporal] scales.</strong> Community ecology underpins the vast fields of biodiversity and biogeography and concerns spatial scales from squares of meters to all of Earth. We can look at historical, contemporary, and future processes that have been implicated in shaping the distribution of life on our planet.</p>
<p>Ecologists tend to analyse how multiple environmental factors act as drivers that influence the distribution of tens or hundreds of species. These data tend to often be messy and statistical considerations need to be understood within the context of the data available to us.</p>
<p><strong>Macroecology</strong> is an all-encompassing view of ecology which seeks to find the geographical patterns and processes in biodiversity across all spatial scales, from local to global, across time scales from years to millennia, and across all taxonomic hierarchies (from genetic variability within species, up to major higher level taxa, such as families and orders). It attempts to arrive a unifying theory for ecology across all of these scales—e.g.&nbsp;one that can explain all patterns in structure and functioning from microbes to blue whales. Most importantly, perhaps, is that it attempts to offer mechanistic explanations for these patterns. At the heart of all explanation is also deep insights stemming from understanding evolution (facilitated by the growth of phylogenetic datasets—see below). This is a modern development of ecology, whereas up to 20 years ago the focus has been mostly on populations (the dynamics of individuals of one species interacting among each other and with their environment) and communities (collections of multiple populations, and how they interact with each other and their environment, and how this affects the structure and dynamics of ecosystems).</p>
<p>On a basic data analytical level, population ecology, community ecology, and macroecology all share the same approach as far as the underlying data are concerned. We start with data representing the species and the associated environmental conditions at a selection of sites (called <strong>species tables</strong> and <strong>environmental tables</strong>). The species tables are then converted to <strong>dissimilarity matrices</strong> and the environmental tables to <strong>distance matrices</strong>. From here basic analyses can offer insights into how biodiversity is structured, e.g.&nbsp;<strong>species-abundance distributions</strong>, <strong>occupancy-abundance curves</strong>, <strong>species-area curves</strong>, <strong>distance decay curves</strong>, and <strong>gradient analyses</strong> (as seen in Shade et al.&nbsp;2018). In the Labs we will explore some of these properties.</p>
</section>
<section id="ecological-data" class="level2">
<h2 class="anchored" data-anchor-id="ecological-data">Ecological data</h2>
<section id="properties-of-ecological-datasets" class="level3">
<h3 class="anchored" data-anchor-id="properties-of-ecological-datasets">Properties of ecological datasets</h3>
<p>Ecological data capture properties of the environment and properties about communities. They are typically stored as separate datasets but they analysed together.</p>
<p>These data sets are usually arrange in a <strong>matrix</strong>, which—in the case of community composition—has <strong>species (or higher level taxa whose resolution depends on the research question at hand) arranged down columns</strong> and <strong>samples (typically the sites, stations, transects, time, plots, etc.) along rows</strong>. We call this a <strong>sites × species table</strong>. In the case of environmental data it will of course be a <strong>sites × environment table</strong>. The term ‘sample’ denote the basic unit of observation. Samples may be quadrats, transects, stations, sites, traps, seine net tows, trawls, grids cells on a map, etc. It is important to be clear and unambiguous about the basic unit of the samples.</p>
<p>A very clear example of environmental and species datasets is the Doubs River dataset. Please refer to <a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:doubs">David Zelený’s website for an explanation</a> of these data. These data demonstrate nicely how one of the basic drivers of biodiversity patterning—gradients—can be seen operating on in a real world case study. It offers neat insight also into the properties of species and environmental tables and the dissimilarity and distance matrices derived from them.</p>
<p>Download the Doubs River data here:</p>
<p></p>

<table>
<thead>
<tr>
<th style="text-align:left">
Dataset
</th>
<th style="text-align:left">
Format
</th>
<th style="text-align:left">
Link
</th>
</tr>
</thead>
<tbody>
</tbody><thead>
<tr>
<th style="text-align:left">
Environmental data
</th>
<th style="text-align:left">
CSV
</th>
<td style="text-align:left">
<a href="/BDC334/data/DoubsEnv.csv"><i class="fas fa-file-csv" title="DoubsEnv.csv"></i></a>
</td>
</tr><tr>
<th style="text-align:left">
Species data
</th>
<th style="text-align:left">
CSV
</th>
<td style="text-align:left">
<a href="/BDC334/data/DoubsSpe.csv"><i class="fas fa-file-csv" title="DoubsSpe.csv"></i></a>
</td>
</tr>
<tr>
<th style="text-align:left">
Spatial data
</th>
<th style="text-align:left">
CSV
</th>
<td style="text-align:left">
<a href="/BDC334/data/DoubsSpa.csv"><i class="fas fa-file-csv" title="DoubsSpa.csv"></i></a>
</td>
</tr>


</thead></table>
<p></p>
</section>
<section id="note-about-csv-files-and-resultant-complications-during-import" class="level3">
<h3 class="anchored" data-anchor-id="note-about-csv-files-and-resultant-complications-during-import">Note about CSV files and resultant complications during import</h3>
<p>These data are made available in CSV format but we can open and view it in MS Excel. ‘CSV’ means <em>comma separated value</em>. It is a plain text file that can be edited in any text editor (such as Notepad on MS Windows, or VS Code, VIM, emacs, etc. on all platforms). The figure shows what a CSV files looks like in a plain text editor, VS Code, on my computer. Once imported it will look similar to the one seen under <a href="/bdc334/chapters/01-introduction/#distances">Distances</a> in the ‘Pair-wise matrices’ section below, but the latter does not have decimals:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/csv_file.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 1.</strong> The content of a CSV file seen in VS Code.</figcaption><p></p>
</figure>
</div>
<p>CSV is a common format used in the scientific disciplines as it is compatible with many software. Globally, scientists use a period ‘.’ as decimal point separator. You can see this in the file above. Commas are used exclusively as field separators (separating the columns you’ll see once opened in MS Excel). This creates a bit of a problem for South Africans, who are indoctrinated from a young age to use commas as decimal point separators—this is to conform with the regional (South African) expectation that dictates commas be used as decimals. So, when you import a CSV file for the first time you’ll likely see gibberish because your computer will probably be set up to honour the regional (locale) expectation of commas as decimal points (and ‘R’ for currency, metric units of measurements, etc.). So, you need to know how to fix this to prevent upsetting me (it is a pet peeve and frustrates me endlessly) and yourselves.</p>
<p>To fix this annoyance is not too tricky as is demonstrated <a href="https://www.avantixlearning.ca/microsoft-excel/how-to-change-commas-to-decimal-points-and-vice-versa-in-excel/">here</a>. Follow the instruction under ‘<strong>Changing commas to decimals and vice versa by changing Excel Options</strong>’. Better still, change the global system settings as explained elsewhere in the same article. Do this before importing the CSV file.</p>
</section>
<section id="looking-at-the-files-content" class="level3">
<h3 class="anchored" data-anchor-id="looking-at-the-files-content">Looking at the files’ content</h3>
<p>After importing the Doubs River data we see something that resembles the next two figures. First, in <code>DoubsSpe.csv</code>, we see the table (or spreadsheet) view of the species data. The species codes for 27 species of fish appear as column headers (not all species’ data are visible as the data are truncated to the right) and in rows 2 through 31 (30 rows) are each of the samples—in this case, there is one sample per site down the length of the river.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/DoubsSpe_table.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 2.</strong> The Doubs River species data in MS Excel.</figcaption><p></p>
</figure>
</div>
<p><code>DoubsEnv.csv</code> contains the environmental data as seen in the next figure. The names of the 11 environmental variables appear as column headers and there are 30 rows, one for each of the samples—the samples match that of the species data.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/DoubsEnv_table.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 3.</strong> The Doubs River environmental data.</figcaption><p></p>
</figure>
</div>
<p>Species data may be recorded as various kinds of measurements, such as presence/absence data, biomass, frequency, or abundance. ‘Presence/absence’ of species simply tells us the the species is there or is not there. It is binary. ‘Abundance’ generally refers to the number of individuals per unit of area or volume, or to percent cover. ‘Biomass’ refers to the mass of the species per unit of area or volume. The type of measure will depend on the taxa and the questions under consideration. The important thing to note is that all species have to be homogeneous in terms of the metric used (i.e.&nbsp;all of it as presence/absence, or abundance, or biomass, not mixtures of them). The matrix’s constituent row vectors are considered the species composition for the corresponding sample (i.e.&nbsp;a row runs across multiple columns, and this tells us that the sample is comprised of all the species whose names are given by the column titles—note that in the case of the data in above figures, it is often the case that there are 0s, meaning that not all species are present at some sites). Species composition is frequently expressed in terms of relative abundance; i.e.&nbsp;constrained to a constant total such as 1 or 100%, or biomass, in which case the upper limit might be arbitrary.</p>
<p>The environmental data may be heterogeneous, i.e.&nbsp;the units of measure may differ among the variables. For example, pH has no units, concentration of some nutrient has a unit of (typically) μM, elevation may be in meters, etc. Because these units differ so much, and because they therefore have different magnitudes and ranges, we may need to standarise them. To standardise the data we substract the mean of each column from each data point in the column, and then divide each of the resultant values by the standard deviation of the columns.</p>
</section>
<section id="lab-tasks" class="level3">
<h3 class="anchored" data-anchor-id="lab-tasks">Lab tasks</h3>
<blockquote class="blockquote">
<p><strong>Task 1:</strong> Standardise the environmental data in MS Excel.</p>
</blockquote>
</section>
<section id="properties-of-species-datasets" class="level3">
<h3 class="anchored" data-anchor-id="properties-of-species-datasets">Properties of species datasets</h3>
<p>Many community data matrices share some general characteristics:</p>
<ul>
<li><p>Most species occur only infrequently. The majority of species might typically be represented at only a <strong>few locations</strong> (where they might be quite abundant). Or some species are simply <strong>rare</strong> in the sampled region (i.e.&nbsp;when they are present, they are present at a very low abundance). This results in <strong>sparse matrices</strong>, as we see in the figures above, where the bulk of the entries consists of zeros.</p></li>
<li><p>Ecologists tend to sample a multitude of factors that they think influence species composition, so the matching environmental data set will also have multiple (10s) of columns that will be assessed in various hypotheses about the drivers of species patterning across the landscape. For example, fynbos biomass may be influenced by the fire regime, elevation, aspect, soil moisture, soil chemistry, edaphic features, etc. These datasets are called <strong>multi-dimensional</strong> matrices, with the ‘dimensions’ referring the the many species or environmental variables.</p></li>
<li><p>Even though we may capture a multitude of information about many environmental factors, <strong>the number of important ones is generally quite low</strong>—i.e.&nbsp;a few factors can explain the majority of the explainable variation, and it is our intention to find out which of them is most important.</p></li>
<li><p>Much of the signal may be spurious, i.e.&nbsp;the matrices have <strong>high noise</strong>. Variability is a general characteristic of the data, and this may result in false patterns emerging. This is so because our sampling may capture a huge amount of stochasticity (processes that are entirely non-deterministic), which may mask the real pattern of interest. Imaginative and creative sampling may reveal some of the patterns we are after, but this requires long years of experience and is not something that can easily be taught as part of our module.</p></li>
<li><p>There is a huge amount of <strong>collinearity</strong>. Basically, what this means is that although it is often the case that many explanatory variables are able to explain patterning, only a few of them act in a way that implies causation. Collinearity is something we will return to later on.</p></li>
</ul>
</section>
</section>
<section id="ecological-gradients" class="level2">
<h2 class="anchored" data-anchor-id="ecological-gradients">Ecological gradients</h2>
<p>Although there are many ways in which species can respond to their environment, one of the most striking responses can be seen along environmental gradients. We will explore this concept next as we discuss coenoclines and unimodal species distribution models.</p>
<section id="coenoclines-coenoplanes-and-coenospaces" class="level3">
<h3 class="anchored" data-anchor-id="coenoclines-coenoplanes-and-coenospaces">Coenoclines, coenoplanes, and coenospaces</h3>
<p>A <strong>coenocline</strong> is a graphical display of <em>all species</em> response curves (see definition, below) <em>simultaneously</em> along one environmental gradient. It aids our understanding of the species response curve if we imagine the gradient operating in only one geographical direction. The <strong>coenoplane</strong> concept extends the coenocline to cover two gradients. Again, our visual representation can be facilitated if the two gradients are visualised orthogonal (in this case, at right angles) to each other (e.g.&nbsp;east–west and north–south) and they do not interact. A <strong>coenospace</strong> complicates the model substantially, as it can allow for an unspecified number of gradients to operate simultaneously on multiple species at once, and it will probably also capture interactions of environmental drivers on the species.</p>
<p>Here is an example of a coenocline using simulated species data. It demonstrates an important idea: that of unimodal species distributions (see below).</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="01-lab_1_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 4. A coenocline.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>A coenoplane is demonstrated next:</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="01-lab_1_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 5. A coenoplane with smooth surface showing idealised species responses.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>Above we see idealised surfaces (smooth models) and the ‘raw’ species counts are obscured. Plotting the actual count data would look more messy, like this:</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="01-lab_1_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
<p></p><figcaption class="figure-caption">Figure 6. A coenoplane with ‘smooth surface’raw’ abundance data.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<section id="species-response-curves" class="level4">
<h4 class="anchored" data-anchor-id="species-response-curves">Species response curves</h4>
<p>Plotting the abundance of a species as a function of position along a gradient is called a <strong>species response curve</strong>. If a long enough gradient is sampled, a species typically has a <em>unimodal</em> response (one peak <em>resembling</em> a Gaussian distribution) to the gradient. Although the idealised Gaussian response is desired (for statistical purposes, largely), in nature the curve might deviate quite noticeably from what’s considered ideal. It is probable that a perfect normal species distribution along a gradient can only be expected when the gradient is perfectly linear in magnitude (seldom true in nature), operates along only one geographical direction (unlikely), and that all other potentially additive environmental influences are constant across the ecological (coeno-) space (also not a realistic expectation).</p>
</section>
<section id="unimodal-species-response" class="level4">
<h4 class="anchored" data-anchor-id="unimodal-species-response">Unimodal species response</h4>
<p>The <strong>unimodal</strong> model is an idealised species response curve (visualised as a coenocline) where a species has only one mode of abundance. The unimodal model offers a convenient heuristic tool for understanding how species can become structured along environmental gradients. In this type of species response curve the species has one optimal environmental condition where it is most abundant (fewest ecophysiological and ecological stressors). If any aspect of the environment is suboptimal (greater or lesser than the optimum), the species will perform more poorly and it will have a lower abundance.</p>
</section>
</section>
</section>
<section id="exploring-data" class="level2">
<h2 class="anchored" data-anchor-id="exploring-data">Exploring data</h2>
<section id="in-the-beginning" class="level3">
<h3 class="anchored" data-anchor-id="in-the-beginning">In the beginning</h3>
<p>At the start of the analysis, before we go deeper into seeing the patterns in the data, we need to explore the data and compute the various synthetic descriptors. This might involve calculating means and standard deviations for some of the variables we feel are most important. So, we say that we produce univariate summaries, and if there is a need. Be guided by the research questions as to what is required. Typically, I don’t like to produce too many detailed inferential statistics of the univariate data (there are special statistical techniques available that allow us to do so more efficiently and effectively, but we will get to it in the Honours Module <a href="/quantecol/">Quantitative Ecology</a>), choosing in stead to see which relationships and patterns emerge from the exploratory summary plots before testing their statistical significance using multivariate approaches. But that is me. Sometimes, some hypotheses call for a few univariate inferential analyses (again, this is the topic of an Honours module on <a href="/workshops/biostats/">Biostatistics</a>).</p>
</section>
<section id="lab-tasks-1" class="level3">
<h3 class="anchored" data-anchor-id="lab-tasks-1">Lab tasks</h3>
<blockquote class="blockquote">
<p><strong>Task 2:</strong> Create an <em>x</em> − <em>y</em> plot of the geographical coordinates in <code>DoubsSpa.csv</code>.</p>
<p><strong>Task 3:</strong> Using some graphs that plot the trends of the environmental variables along the length of the river, describe the patterns in some of the environmental variables and offer explanations for how they might be responsible for affecting species distributions down the length of the Doubs River. Which three variable do you think will be be able to explain the trends in the species data?</p>
</blockquote>
</section>
</section>
<section id="pair-wise-matrices" class="level2">
<h2 class="anchored" data-anchor-id="pair-wise-matrices">Pair-wise matrices</h2>
<p>Although we typically start our forays into data exploration using sites × species and sites × environment tables, the formal statistical analyses usually require ‘pairwise association matrices.’ Such matrices are symmetrical (or lower triangle) square matrices (i.e.&nbsp;<em>n</em> × <em>n</em>). These matrices tell us about how related any sample is with any other sample in our pool of samples (i.e.&nbsp;relatedness among rows with respect to whatever populates the columns, be they species information of environmental information).</p>
<p>Let us consider various kinds of distance matrices under the headings <strong>Distances</strong>, <strong>Correlations</strong>, <strong>Associations</strong>, <strong>Similarities</strong>, and <strong>Dissimilarities</strong>.</p>
<section id="distances" class="level3">
<h3 class="anchored" data-anchor-id="distances">Distances</h3>
<p>A frequently used <strong>distance metric</strong> is Euclidean <strong>distance</strong>. Euclidean distance is the ‘ordinary straight-line’ distance between two points in Euclidean space. Working with geographical coordinates over small areas of Earth’s surface, Euclidean distance is very similar (i.e. almost directly proportional) to the actual geographical distance, so the concept is quite intuitive to understand. In its simplest form it is a planar Cartesian area, which you know of as a graph with <em>x</em>- and <em>y</em>-axes). So, in 2D and 3D space it gives distances in Cartesian units between points on plane <em>x</em>, <em>y</em> or in volume <em>x</em>, <em>y</em>, <em>z</em>; a linear relationship exists between the units in the physical realm and the units in Euclidean space. The implication is that on a map or on a graph, short distances between pairs of points indicate that there are also a short geographic distances between these points on Earth.</p>
<p>Euclidean distance is calculated using the Pythagorean theorem and it is typically applied to standardised environmental data (not species data):</p>
<p><span class="math display">\[d(a,b) = \sqrt{(a_x - b_x)^2 + (a_y - b_y)^2 + (a_z - b_z)^2} \]</span></p>
<p>In the above equation, this ‘distance’ is calculated between a pair of sites, <strong><em>a</em></strong> and <strong><em>b</em></strong>, whose locations are marked by the coordinates <em>x</em>, <em>y</em>, and <em>z</em>—i.e.&nbsp;this is an example of 3-dimensional data (a space or volume, as opposed to 2D data situated on a <em>x</em>, <em>y</em> place). We might also call each coordinate a ‘variable’ such as temperature, depth, or light intensity (sometimes also called ‘dimensions’ of environmental space).</p>
<p><span class="math display">\[d(a,b) = \sqrt{(a\_{temp} - b\_{temp})^2 + (a\_{depth} - b\_{depth})^2 + (a\_{light} - b\_{light})^2} \]</span></p>
<p>Let’s <a href="../data/Euclidian_distance_demo_data_xyz.csv">download some example data</a> and calculate the distance between every pair of sites named <strong><em>a</em></strong> to <strong><em>g</em></strong>.</p>
<p>The ‘raw’ data representing <em>x</em>, <em>y</em> and <em>z</em> dimensions look like this in MS Excel:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Euclidian_xyz.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 7.</strong> Three-dimensional data at seven sites.</figcaption><p></p>
</figure>
</div>
<p>Or we can substitute <em>x</em>, <em>y</em> and <em>z</em> for environmental ‘dimensions’ and we have this:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Euclidian_env.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 7.</strong>’Three-dimensional’ environmental data at seven sites.</figcaption><p></p>
</figure>
</div>
<p>Regardless of whether we have <em>x</em>, <em>y</em> and <em>z</em> or environmental dimensions, the application of the Theorem of Pythagoras is the same.</p>
<p>We can calculate Euclidean distance in Excel using some built-in functions:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Euclidian_env_calc.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 8.</strong> Calculating Euclidean distances in MS Excel.</figcaption><p></p>
</figure>
</div>
</section>
<section id="correlations" class="level3">
<h3 class="anchored" data-anchor-id="correlations">Correlations</h3>
<p>We use correlations to establish the extent to which environmental variables relate to one another across the sample sites. Note, therefore, that correlations performed to a sites × variable table are done between columns (variables), not rows as in the Euclidean distance calculation, which compares the rows (sites). We do not need to standardise as one would do for the calculation of Euclidean distances. Correlation coefficients (so called <em>r</em>-values) vary in magnitude from -1 (a perfect inverse relationship) through 0 (no relationship whatsoever) to 1 (a perfect positive linear relationship).</p>
<p>The resultant pairwise correlation matrix shows the names of the environmental variables as both column and row names. Contrast this with what is presented as row and column names in the distance matrix.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/Correlation_env_calc.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption"><strong>Figure 9.</strong> Calculating pairwise correlations in MS Excel.</figcaption><p></p>
</figure>
</div>
</section>
<section id="lab-tasks-2" class="level3">
<h3 class="anchored" data-anchor-id="lab-tasks-2">Lab tasks</h3>
<blockquote class="blockquote">
<p><strong>Task 4:</strong> Using the Doubs River environmental data, calculate the lower left triangle (including the diagonal) distance matrix for <em>every pair</em> of sites in Sites 1, 3, 5, …, 29 (i.e.&nbsp;using only every second site). Explain any patterns or trends in this resultant distance matrix in terms of how similar/different sites are relative to each other. Which of the graphs you came up with in Task 3 (if any) do you think are most responsible for the patterns seen in the distance matrix?</p>
<p><strong>Task 5:</strong> Using the same sites as above (Task 4), calculate a pairwise correlation matrix (lower left and including the diagonal) for the Doubs River environmental data. Explain any patterns or trends in this resultant correlation matrix and offer mechanistic explanations for why these correlations might exist.</p>
<p><strong>Task 6:</strong> Discuss in detail the properties of distance and correlation matrices.</p>
<p><strong>Task 7:</strong> If you found this exercise annoying, explain why. What could be done to ease your experience of the calculations? Or if you loved it, state why.</p>
</blockquote>
</section>
<section id="associations-similarities-and-dissimilarities" class="level3">
<h3 class="anchored" data-anchor-id="associations-similarities-and-dissimilarities">Associations, similarities, and dissimilarities</h3>
<p>We will discuss and calculate association, similarity and dissimilarity matrices in Lab 2.</p>
<!-- Two samples with similar species composition are ecologically similar, while two samples that share few species are ecologically distant. In Figure 4.1, below, the data displayed in Figure 2.1 have been converted into a dissimilarity distance matrix of dimension $30 \times 27$. The are a few things of interest in this matrix: -->
<!-- * The distance matrix is square and therefore symmetrical. In other words, there are as many rows as there are columns, and this number corresponds to the number of samples in our sites × species matrix. -->
<!-- * The cells of the diagonal running from top-left to bottom-right contain zeros, showing rather obviously that there is no difference between a sample and the sample itself.  -->
<!-- * The 'upper triangle' above the diagonal is an inversion of the 'lower triangle' below the diagonal; because they are identical in terms of the pairwise relationships that they encode, distance matrices are sometimes represented simply by the lower triangular matrix. -->
<!-- * These matrices contain ecological information. For example, between samples (here each of 58 × 50-km long coastal sections) that are geographically close together, the dissimilarity will is generally low (i.e. the samples are similar in their species composition), while the further sites are removed from each other, the greater the dissimilarity will be. (*Note: not all samples are not always related to each other as a function of distance --- this is a characteristic of the data used for this particular example analysis, so be aware of the context when interpreting distance matrices.*) -->
<!-- * All information about the particular species present within each sample are now gone since that information has been collapsed to a dissimilarity measure. -->
</section>
</section>
<section id="take-note" class="level2">
<h2 class="anchored" data-anchor-id="take-note">Take note</h2>
<p>Provide a <strong>neat and thoroughly annotated</strong> MS Excel spreadsheet which outlines the graphs and all calculations, and which displays the resultant distance matrix. Use separate tabs for the different questions. Written answers must be typed in a MS Word document. Please make sure you follow the formatting specifications <em>precisely</em> as shown in the file <strong>BDC334 Example essay format.docx</strong> that was circulated at the beginning of the module. Feel free to use the file as a template.</p>
<p>Please label the MS Excel and MS Word files as follows: **BDC334_<first_name>_<last_name>_Assignment_1.xlsx** and **BDC334_<first_name>_<last_name>_Assignment_1.docx** (the ‘&lt;’ and ‘&gt;’ must be omitted as they are used in the example as field indicators only).</last_name></first_name></last_name></first_name></p>
<p>Failing to follow these instructions carefully, precisely, and thoroughly will cause you to lose marks, which could cause a significant drop in your score as formatting counts for 15% of the final mark (out of 100%).</p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>